micall/core/aln2counts.py

```diff
@@ -1358,74 +1504,123 @@ def add_nuc_read(self, offset_sequence, count):
+                    if is_valid:
+                        current_nuc_counts[insert_nuc_seq] += count
+
+            for left, counts in insert_nuc_counts.items():
+                pos = insert_behind.get(left)
+                if pos is None:
+                    # this can happen if the insertion is at the end of an alignment,
+                    # because we only look at the consensus on the right edge of the insert
+                    continue
+                ref_pos = pos - 1
+                self.ref_insertions[region][ref_pos] =\
+                    aggregate_insertions(insert_nuc_counts[left],
+                                         consensus_pos=left-1,
+                                         coverage_nuc=insertion_coverage[left])
+                insert_pos = insert_behind.get(left) // 3
+                count = sum(counts.values())
                 # Only care about insertions in the middle of the sequence,
                 # so ignore any that come before or after the reference.
                 # Also report if we're in test mode (no report_aminos).
-                if not report_aminos or insert_before not in (1, None):
-                    row = dict(seed=self.seed,
-                               region=region,
-                               qcut=self.qcut,
-                               left=left + 1,
-                               insert=insert_seq,
-                               count=count,
-                               before=insert_before)
-                    self.insert_writer.writerow(row)
-                    if insert_before is not None:
-                        region_insert_pos_counts[insert_before-1] += count
+                if not report_aminos or insert_pos not in (0, None):
+                    if insert_pos is not None:
+                        region_insert_pos_counts[insert_pos] += count
+
+        self.parse_conseq_insertions(seed_name, report_nucleotides_all)
+        self.write_insertions_file(landmarks, consensus_builder)
+
+    def parse_conseq_insertions(self, seed_name, report_nucleotides_all):
+        for insertion_position in self.conseq_insertions[seed_name].keys():
+            insertions = self.conseq_insertions[seed_name][insertion_position]
+            for region in report_nucleotides_all:
+                report_aminos = []
+                report_nucleotides = report_nucleotides_all[region]
+                current_insert_behind, current_insert_coverage = \
+                    get_insertion_info(insertion_position, report_aminos, report_nucleotides)
+                if current_insert_behind is not None:
+                    for position in insertions:
+                        insertions[position].count_nucleotides('-', count=current_insert_coverage)
+                    if self.ref_insertions[region] is None:
+                        self.ref_insertions[region] = defaultdict(lambda: defaultdict(lambda: SeedNucleotide()))
```

Comment:
Another unneeded lambda, and this line isn't covered by a test.

Reply:
Because `self.ref_insertions` is already a defaultdict, I don't think we need the line at all - I've taken it out.